ql/src/java/org/apache/hadoop/hive/ql/Driver.java

```diff
@@ -213,7 +216,12 @@ private void runInternal(String command, boolean alreadyCompiled) throws Command
         releaseResources();
       }
 
+      if (SessionState.get() != null) {
+        // Remove any query state reference from the session state
+        SessionState.get().removeQueryState(getConf().get(HiveConf.ConfVars.HIVEQUERYID.varname));
+      }
       driverState.executionFinishedWithLocking(isFinishedWithError);
+
     }
```

Comment:
nit of the nit for the newlines:
```suggestion
      }

      if (SessionState.get() != null) {
        // Remove any query state reference from the session state
        SessionState.get().removeQueryState(getConf().get(HiveConf.ConfVars.HIVEQUERYID.varname));
      }

      driverState.executionFinishedWithLocking(isFinishedWithError);
    }
```